src/Npgsql.GeoJSON/GeoJsonHandler.cs

```diff
@@ -0,0 +1,796 @@
+                for (var i = 1; i < coordinates.Count; ++i)
+                    if (HasZ(coordinates[i]) != hasZ) return true;
+            }
+            return false;
+        }
+
+        public int ValidateAndGetLength(MultiPoint value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+        {
+            var length = SizeOfHeaderWithLength;
+            if (value.CRS != null)
+                length += sizeof(int);
+
+            var coordinates = value.Coordinates;
+            for (var i = 0; i < coordinates.Count; ++i)
+                length += ValidateAndGetLength(coordinates[i], ref lengthCache, parameter);
+
+            return length;
+        }
+
+        public int ValidateAndGetLength(MultiLineString value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+        {
+            var length = SizeOfHeaderWithLength;
+            if (value.CRS != null)
+                length += sizeof(int);
+
+            var coordinates = value.Coordinates;
+            for (var i = 0; i < coordinates.Count; ++i)
+                length += ValidateAndGetLength(coordinates[i], ref lengthCache, parameter);
+
+            return length;
+        }
+
+        public int ValidateAndGetLength(MultiPolygon value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+        {
+            var length = SizeOfHeaderWithLength;
+            if (value.CRS != null)
+                length += sizeof(int);
+
+            var coordinates = value.Coordinates;
+            for (var i = 0; i < coordinates.Count; ++i)
+                length += ValidateAndGetLength(coordinates[i], ref lengthCache, parameter);
+
+            return length;
+        }
+
+        public int ValidateAndGetLength(GeometryCollection value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+        {
+            var length = SizeOfHeaderWithLength;
+            if (value.CRS != null)
+                length += sizeof(int);
+
+            var geometries = value.Geometries;
+            for (var i = 0; i < geometries.Count; ++i)
+                length += ValidateAndGetLength((GeoJSONObject)geometries[i], ref lengthCache, parameter);
+
+            return length;
+        }
+
+        int INpgsqlTypeHandler<IGeoJSONObject>.ValidateAndGetLength(IGeoJSONObject value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+            => ValidateAndGetLength((GeoJSONObject)value, ref lengthCache, parameter);
+
+        int INpgsqlTypeHandler<IGeometryObject>.ValidateAndGetLength(IGeometryObject value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+            => ValidateAndGetLength((GeoJSONObject)value, ref lengthCache, parameter);
+
+        public override Task Write(GeoJSONObject value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+        {
+            switch (value.Type)
+            {
+                case GeoJSONObjectType.Point:
```

Comment:
Please unindent case to the same level as the switch

Reply:
Will do.